portal-api/src/v1/validation/checker-submit-deal-facility-ready-for-check-cover-start-dates/index.js

```diff
@@ -3,21 +3,36 @@ const { orderNumber } = require('../../../utils/error-list-order-number');
 const { formattedTimestamp } = require('../../facility-dates/timestamp');
 const CONSTANTS = require('../../../constants');
 
-const facilityReadyForCheckRequestedCoverStartDateRule = (facility, errorList) => {
+const facilityReadyForCheckRequestedCoverStartDateRule = (deal, facility, errorList) => {
   const {
     requestedCoverStartDate,
     status,
   } = facility;
+  const { details, submissionType } = deal;
+  const { submissionDate, manualInclusionApplicationSubmissionDate, manualInclusionNoticeSubmissionDate } = details;
+  let dateOfSubmission;
   const newErrorList = errorList;
 
+  if (submissionDate || manualInclusionApplicationSubmissionDate || manualInclusionNoticeSubmissionDate) {
+    if (submissionType === CONSTANTS.DEAL.SUBMISSION_TYPE.AIN) {
+      dateOfSubmission = submissionDate;
+    } else if (manualInclusionNoticeSubmissionDate) {
+      dateOfSubmission = manualInclusionNoticeSubmissionDate;
+    } else {
+      dateOfSubmission = manualInclusionApplicationSubmissionDate;
```

Comment:
* During first application submission as an `MIA` `manualInclusionApplicationSubmissionDate` and `submissionDate` might be empty will that be the case when submitting from checker to UKEF?
* If above is true then what value will `dateOfSubmission` have?
* Will `manualInclusionNoticeSubmissionDate` be available on submission to UKEF or will the property be empty?
* When submitting `GEF` will this function be triggered, if yes then need to ensure correct MIN submissions date is picked.